src/modules/ens/sagas.ts

```diff
@@ -207,3 +217,43 @@ function* handleFetchENSListRequest(_action: FetchENSListRequestAction) {
     yield put(fetchENSListFailure(ensError))
   }
 }
+
+function* handleChangeProfile(action: ChangeProfileRequestAction) {
+  try {
+    const { selectedName } = action.payload
+
+    const address = yield select(getAddress)
+
+    const client = new CatalystClient('https://peer.decentraland.org', 'builder')
+    const entities = yield client.fetchEntitiesByPointers(EntityType.PROFILE, [address.toLowerCase()])
+    const profile = entities.length > 0 ? entities[0] : null
+
+    const avatar = profile && profile.metadata && profile.metadata.avatars[0]
+    avatar.name = selectedName
+    avatar.hasClaimedName = true
+
+    // Build entity
+    const content: Map<string, string> = new Map(
+      (profile.content || []).map(({ file, hash }: { file: string; hash: string }) => [file, hash])
+    )
+    const deployPreparationData = yield call(() =>
+      DeploymentBuilder.buildEntityWithoutNewFiles(EntityType.PROFILE, [address], content, profile.metadata)
+    )
+
+    // Request signature
+    const eth = Eth.fromCurrentProvider()
+    if (eth) {
+      const personal = new Personal(eth.provider)
+      const signature = yield personal.sign(deployPreparationData.entityId, address, '')
+
+      // Deploy change
+      const authChain = Authenticator.createSimpleAuthChain(deployPreparationData.entityId, address, signature)
+      yield client.deployEntity({ ...deployPreparationData, authChain })
+
+      yield put(changeProfileSuccess())
+    }
+  } catch (error) {
+    const ensError: ENSError = { message: error.message }
+    yield put(changeProfileFailure(ensError))
+  }
+}
```

Comment:
As a general comment we talked about in another PR. `yield`ing promises technically works, but the preferred method is to use `call`, the helper provided by redux-sagas. So code could be rewritten like this:
```diff
- yield client.deployEntity({ ...deployPreparationData, authChain })
+ yield call(() => client.deployEntity({ ...deployPreparationData, authChain }))
```

check other sites of the same occurence in this file like `yield call(() => ensContract.methods.resolver(nodehash).call())`

Reply:
Done and Fixed!